docs/appendix-A.rst

```diff
@@ -0,0 +1,1783 @@
+   :end-before: end STR_pass_char_ptr
+
+The C wrapper:
+
+.. literalinclude:: ../regression/reference/strings/wrapstrings.cpp
+   :language: c
+   :start-after: start STR_pass_char_ptr_bufferify
+   :end-before: end STR_pass_char_ptr_bufferify
+
+Calls C via the interface:
+
+.. literalinclude:: ../regression/reference/strings/wrapfstrings.f
+   :language: fortran
+   :start-after: start c_pass_char_ptr
+   :end-before: end c_pass_char_ptr
+   :dedent: 8
+
+.. literalinclude:: ../regression/reference/strings/wrapfstrings.f
+   :language: fortran
+   :start-after: start c_pass_char_ptr_bufferify
+   :end-before: end c_pass_char_ptr_bufferify
+   :dedent: 8
+
+The Fortran wrapper:
+
+.. literalinclude:: ../regression/reference/strings/wrapfstrings.f
+   :language: fortran
+   :start-after: start pass_char_ptr
+   :end-before: end pass_char_ptr
+   :dedent: 4
+
+The function can be called without the user aware that it is written in C++:
+
+.. code-block:: fortran
+
+    character(30) str
+    call pass_char_ptr(dest=str, src="mouse")
+
+
+.. ############################################################
+
+.. _example_ImpliedTextLen:
+
+ImpliedTextLen
+^^^^^^^^^^^^^^
+
+Pass the pointer to a buffer which the C library will fill.  The
+length of the buffer is passed in ``ltext``.  Since Fortran knows the
+length of ``CHARACTER`` variable, the Fortran wrapper does not need to
+be explicitly told the length of the variable.  Instead its value can
+be defined with the *implied* attribute.
+
+This can be used to emulate the behavior of most Fortran compilers
+which will pass an additional, hidden argument which contains the
+length of a ``CHARACTER`` argument.
+
+.. literalinclude:: ../regression/run/clibrary/clibrary.c
+   :language: c
+   :start-after: start ImpliedTextLen
+   :end-before: end ImpliedTextLen
+
+YAML:
+
+.. code-block:: yaml
+
+    - decl: void ImpliedTextLen(char *text+intent(out)+charlen(MAXNAME),
+                                int ltext+implied(len(text)))
+
+The C wrapper:
+
+.. literalinclude:: ../regression/reference/clibrary/wrapClibrary.c
+   :language: c
+   :start-after: start CLI_implied_text_len_bufferify
+   :end-before: end CLI_implied_text_len_bufferify
+
+Calls C via the interface:
+
+.. literalinclude:: ../regression/reference/clibrary/wrapfclibrary.f
+   :language: fortran
+   :start-after: start c_implied_text_len_bufferify
+   :end-before: end c_implied_text_len_bufferify
+   :dedent: 8
+
+The Fortran wrapper:
+
+.. literalinclude:: ../regression/reference/clibrary/wrapfclibrary.f
+   :language: fortran
+   :start-after: start implied_text_len
+   :end-before: end implied_text_len
+   :dedent: 4
+
+Fortran usage:
+
+.. code-block:: fortran
+
+    character(MAXNAME) name1
+    call implied_text_len(name1)
+    call assert_equals("ImpliedTextLen", name1)
+
+
+std::string
+-----------
+
+.. ############################################################
+
+.. _example_acceptStringReference:
+
+acceptStringReference
```

Comment:
I'm assuming this would be similar for `const std::string& str``.

Also, is it worth mentioning if `std::string` is special cased in shroud.

```cpp
void acceptStringReference(std::string & arg1)
{
    arg1.append("dog");
}
```